### PR TITLE
Fixes aggregation of image datasets

### LINKED ISSUE #2715: `aggregate_datasets` loses Image feature schema for image datasets
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
- LeRobot version: 0.4.3
- Platform: macOS-15.6.1-arm64-arm-64bit
- Python version: 3.10.13
- Huggingface Hub version: 0.34.3
- Datasets version: 4.1.1
- Numpy version: 2.2.6
- FFmpeg version: 7.1.1
- PyTorch version: 2.7.1
- Is PyTorch built with CUDA support?: False
- Cuda version: N/A
- GPU model: N/A
- Using GPU in script?: NO
- lerobot scripts: ['lerobot-calibrate', 'lerobot-dataset-viz', 'lerobot-edit-dataset', 'lerobot-eval', 'lerobot-find-cameras', 'lerobot-find-joint-limits', 'lerobot-find-port', 'lerobot-imgtransform-viz', 'lerobot-info', 'lerobot-record', 'lerobot-replay', 'lerobot-setup-motors', 'lerobot-teleoperate', 'lerobot-train']
```

### Description

When aggregating image-based datasets using aggregate_datasets, the resulting parquet files lose the HuggingFace Image feature type. Image columns are written with a generic struct schema `({'bytes': Value('binary'), 'path': Value('string')})` instead of the proper `Image()` feature type. In turn, the aggregated dataset becomes difficult to use because the features in the parquet file are dropped and bytes are written. Visualizations on the hub are also faulty. Same applies to merge datasets.

This directly causes downstream issues when loading the aggregated dataset (from personal experience, 😅)

## Root cause

The current `to_parquet_with_hf_images` function creates a `datasets.Dataset` without specifying the features schema:

```python
def to_parquet_with_hf_images(df: pandas.DataFrame, path: Path) -> None:    datasets.Dataset.from_dict(df.to_dict(orient="list")).to_parquet(path)
```
When features is not provided to `Dataset.from_dict()`, `datasets` infers the schema from the data (internally stored as `{'bytes': ..., 'path': ...}` dicts), these get typed as a generic struct rather than the `Image()` feature type.

### Relevant logs or stack trace

```Shell

```

### Checklist

- [x] I have searched existing tickets to ensure this isn't a duplicate.
- [x] I am using the latest version of the `main` branch.
- [x] I have verified this is not an environment-specific problem.

### Additional Info / Workarounds

(opening a PR solving this issue right now, 😉)

### PR DESCRIPTION
**Pull request by fracapuano**
## Title
Fixes aggregation of image datasets

## Type / Scope

- **Type**: Bug
- **Scope**: Bug fix for `aggregate_datasets`. Also affects `lerobot-edit-dataset`

## Summary / Motivation
- Fixes #2715. Also, adds tests to ensure this edge case is properly covered.
- As a side note: I'd advocate for removing image datasets altogether :)

## Related issues

- Fixes / Closes: #2715 

## What changed

- Short, concrete bullets of the modifications (files/behaviour).
- Short note if this introduces breaking changes and migration steps.

## How was this tested

- Tests added: list new tests or test files.

## How to run locally (reviewer)

- Run the relevant tests:

  ```bash
  pytest tests/datasets/test_aggregate.py::test_aggregate_image_datasets
  ```

Run these tests to confirm no breaking changes on closely related parts of the library.

```bash
pytest tests/datasets/test_aggregate.py
========================================== test session starts ===========================================
platform darwin -- Python 3.10.13, pytest-8.4.1, pluggy-1.6.0
rootdir: /Users/fracapuano/Desktop/personal/lerobot
configfile: pyproject.toml
plugins: timeout-2.4.0, anyio-4.10.0, cov-6.2.1, mock-serial-0.0.1, hydra-core-1.3.2
collected 4 items                                                                                        

Testing with DEVICE='cpu'

tests/datasets/test_aggregate.py ....                                                              [100%]

=========================================== 4 passed in 15.24s ===========================================
(lerobot) ➜  lerobot git:(fix/fracapuano-aggregate-image-datasets) ✗ pytest tests/datasets/test_dataset_tools.py
========================================== test session starts ===========================================
platform darwin -- Python 3.10.13, pytest-8.4.1, pluggy-1.6.0
rootdir: /Users/fracapuano/Desktop/personal/lerobot
configfile: pyproject.toml
plugins: timeout-2.4.0, anyio-4.10.0, cov-6.2.1, mock-serial-0.0.1, hydra-core-1.3.2
collected 40 items                                                                                       

Testing with DEVICE='cpu'

tests/datasets/test_dataset_tools.py ........................................                      [100%]

========================================== 40 passed in 44.80s ===========================================
```

## Checklist (required before merge)

- [x] Linting/formatting run (`pre-commit run -a`)
- [x] All tests pass locally (`pytest`) (run test_aggregate.py, test_dataset_tools.py)
- **NA** [] Documentation updated
- [x] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

File: src/lerobot/datasets/aggregate.py [hunk 1]
```diff
@@ -527,12 +541,17 @@ def append_or_create_parquet_file(
         final_df = df
         target_path = new_path
     else:
-        existing_df = pd.read_parquet(dst_path)
+        if contains_images:
+            # Use HuggingFace datasets to read existing data to preserve image format
+            existing_ds = datasets.Dataset.from_parquet(str(dst_path))
```
[the remark this answers is not in the document]
Comment (fracapuano) [reply src/lerobot/datasets/aggregate.py:546]: I don't think this is accurate actually---my 2 cents are that this would be an intermediate casting to image features that would be duplicated once again at the time of the final write, hence why the code works even without this single fix. With that being said, I don't have anything against adding it here too. It would feel clearer, but also duplicate/increase unnecessary work (the current PR is fixed and tested already!)
